File: src/main/java/com/lxy/flowplan/pojo/Task.java
package com.lxy.flowplan.pojo;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("task")
public class Task {

    // Task 是 Project 下的最小排期单元，id 由数据库自增生成。
    @TableId(type = IdType.AUTO)
    private Integer id;

    @JsonProperty("project_id")
    private Integer projectId;

    private String title;

    private String description;

    // V1.1 通过权重分配 Project 总预算，不再在 Task 上维护总估时/已完成分钟。
    private Integer weight = 1;

    @JsonProperty("min_session_minutes")
    private Integer minSessionMinutes;

    @JsonProperty("begin_date")
    private LocalDate beginDate;

    private LocalDate deadline;

    @JsonProperty("dependency_task_id")
    private Integer dependencyTaskId;

    // doneFlag 是用户手动完成标记，status 是面向界面的简化状态。
    @JsonProperty("done_flag")
    private Boolean doneFlag = false;

    private String status = "NOT_STARTED";

    // 创建和更新时间由数据库默认值或服务端维护。
    @JsonProperty("created_at")
    private LocalDateTime createdAt;

    @JsonProperty("updated_at")
    private LocalDateTime updatedAt;
}

File: frontend/src/api/task.js
import request from "./request";

function toNumberOrNull(value) {
    if (value === undefined || value === null || value === "") {
        return null;
    }
    return Number(value);
}

function normalizeTask(task) {
    return {
        ...task,
        weight: task.weight ?? 1,
        min_session_minutes: task.min_session_minutes ?? null,
        dependency_task_id: task.dependency_task_id ?? null,
        done_flag: Boolean(task.done_flag),
        status: task.status || (task.done_flag ? "DONE" : "NOT_STARTED")
    };
}

function normalizeTaskPayload(task) {
    return {
        title: task.title?.trim(),
        description: task.description?.trim() || null,
        weight: Number(task.weight || 1),
        min_session_minutes: toNumberOrNull(task.min_session_minutes),
        begin_date: task.begin_date || null,
        deadline: task.deadline || null,
        dependency_task_id: toNumberOrNull(task.dependency_task_id),
        done_flag: Boolean(task.done_flag),
        status: task.status || (task.done_flag ? "DONE" : "NOT_STARTED")
    };
}

export async function listTasks(projectId) {
    const tasks = await request.get(`/project/${projectId}/task/list`);
    return Array.isArray(tasks) ? tasks.map(normalizeTask) : [];
}

export function addTask(projectId, task) {
    return request.post(`/project/${projectId}/task/add`, normalizeTaskPayload(task));
}

export function updateTask(projectId, taskId, task) {
    return request.put(`/project/${projectId}/task/${taskId}`, normalizeTaskPayload(task));
}

export function deleteTask(projectId, taskId) {
    return request.delete(`/project/${projectId}/task/${taskId}`);
}
